docs/developers/style-guide.md

```diff
@@ -2,6 +2,133 @@
 title: Style Guide
 ---
 
-!!! warning
+This section collects general coding style guidelines used in Fast-LLM.
+Following these will ensure a swift reviewing process and will help maintain consistency and readability.
+Note that while we try to enforce these principles,
+exceptions may be allowed on a case-by-case basis, for example if they noticeably improve readability.
 
-    This section is work in progress. Please check back soon for the updated content.
+As a general principle, **Fast-LLM prioritizes code readability and maintainability** over conciseness,
+coding speed or individual programmer's preferences.
+Most of the style choices below are based on this principle.
+
+## 🎯 Basic Style
+
+Unless otherwise specified, Fast-LLM follows the [PEP 8](https://peps.python.org/pep-0008/) coding style.
+This style (and many other conventions) is enforced with automatic formatting through a [pre-commit](https://pre-commit.com/) git hook.
+
+Please make sure these git hooks are installed by running
+
+```bash
+pip install pre-commit
+pre-commit install
+```
```

Comment:
I can confirm that the `pre-commit` command works.

I tested it with:

```
pre-commit run --all-files
```

and

```
pre-commit run --files docs/**/*.md
```